### PR TITLE
Move to using K8s/Openshift modules from Ansible core

### PR DESCRIPTION
**Pull request by ehelms**
The modules we were using have been officially deprecated in favor of these modules in Ansible core as of 2.5.

### CODE REVIEW

**Comment by ehelms** [comment]
[test containers]

**Comment by ehelms** [comment]
@carbonin Any chance you could glance at this?

**Comment by ehelms** [comment]
[test containers]

File: containers/deploy/candlepin/deployments.yml [hunk 1]
```diff
@@ -1,27 +1,38 @@
...
           spec:
+              replicas: 1
+              strategy:
+                resources: {}
+                rollingParams:
+                  intervalSeconds: 1
+                  maxSurge: 25%
+                  maxUnavailable: 25%
+                  timeoutSeconds: 900
+                  updatePeriodSeconds: 1
```
Comment (carbonin) [inline containers/deploy/candlepin/deployments.yml]: Are these things we need because of the new library or separate enhancements?

File: containers/images/foreman/Dockerfile [hunk 2]
```diff
@@ -34,7 +34,6 @@ COPY container-assets/start_foreman.sh /usr/bin/start_foreman.sh
 COPY container-assets/wait_on_postgres.py /usr/bin/wait_on_postgres.py
 COPY container-assets/create_qpid_queue.sh /usr/bin/create_qpid_queue.sh
 
-RUN yum -y reinstall http://koji.katello.org/kojifiles/work/tasks/1420/101420/tfm-rubygem-puma-3.11.4-1.el7.x86_64.rpm
```
Comment (carbonin) [inline containers/images/foreman/Dockerfile]: How is this related to the new ansible module?

File: containers/tools/deploy.yml [hunk 3]
```diff
@@ -12,21 +12,12 @@
...
-    - name: 'Install python openshift'
-      command: pip install git+https://github.com/ehelms/openshift-restclient-python.git@fix-destination-ca
-
-    - name: 'Install kubernetes modules'
-      command: 'ansible-galaxy install -r requirements.yml'
-      args:
-        chdir: "{{ working_dir }}"
+    - name: 'Install openshift'
+      pip:
+        name: openshift
```
Comment (carbonin) [inline containers/tools/deploy.yml]: The ansible module [docs](https://docs.ansible.com/ansible/latest/modules/openshift_raw_module.html#openshift-raw-module) list a rather strict version requirement for this package (`== 0.4.3`). Can we (and do we want to) enforce that here?

File: containers/deploy/candlepin/deployments.yml (hunk 1 above)
Comment (ehelms) [reply containers/deploy/candlepin/deployments.yml]: They were me testing something and copy from Openshift itself. I can drop them.

File: containers/images/foreman/Dockerfile (hunk 2 above)
Comment (ehelms) [reply containers/images/foreman/Dockerfile]: It's not, but builds were breaking without it. I can split to it's down PR.

File: containers/tools/deploy.yml (hunk 3 above)
Comment (ehelms) [reply containers/tools/deploy.yml]: I noticed that too, and I've had no issues with newer versions. Further, newer versions handle some more advanced options better (such as the certificate_ca_file) so I am inclined to ride latest.

File: containers/images/foreman/Dockerfile (hunk 2 above)
Comment (carbonin) [reply containers/images/foreman/Dockerfile]: I think just at least another commit with some details would be good.

File: containers/deploy/foreman-proxy/deployments.yml [hunk 4]
```diff
@@ -9,12 +9,12 @@
     - restart
 
 - name: Create Foreman Proxy deployment
-  openshift_v1_deployment_config:
+  openshift_raw:
       state: "{{ deployment_state }}"
       force: false
       resource_definition:
```
Comment (carbonin) [inline containers/deploy/foreman-proxy/deployments.yml]: Does this need to change?

Comment (carbonin) [reply containers/deploy/foreman-proxy/deployments.yml]: Now that I'm looking for it. It looks like this was missed in quite a few places. Does this need to be changed to `definition` everywhere?

Comment (ehelms) [reply containers/deploy/foreman-proxy/deployments.yml]: Not required, this still works and is supported from what I can tell from the test run.

Comment (carbonin) [reply containers/deploy/foreman-proxy/deployments.yml]: Okay, then I think we should probably be consistent rather than using `definition` in some places and `resource_definition` in others. I'm fine either way, but I think it's strange to have both.

File: containers/deploy/candlepin/deployments.yml [hunk 5]
```diff
@@ -1,27 +1,34 @@
...
           metadata:
               name: candlepin
               labels:
                   app: foreman
                   service: candlepin
               namespace: "{{ project_name }}"
           spec:
+              replicas: 1
+              strategy:
+                resources: {}
```
Comment (carbonin) [inline containers/deploy/candlepin/deployments.yml]: Not sure we need this either.
